tests/test_util_modules.py

```diff
@@ -262,6 +262,7 @@ def _check_result_and_set(future):
 
     future1, future2 = receiver.recv()
     future1.set_result(123)
+    events[6].wait()
```

Comment:
Without this event, we could `assert future2.done()` in the main process before it's cancelled in the child process